src/pyflakes/plugins/repeated_keys.rs

```diff
@@ -0,0 +1,97 @@
+use rustpython_ast::{Constant, Expr, ExprKind};
+
+use crate::ast::cmp;
+use crate::ast::helpers::unparse_expr;
+use crate::ast::types::Range;
+use crate::autofix::Fix;
+use crate::checkers::ast::Checker;
+use crate::registry::{Check, CheckCode, CheckKind};
+
+#[derive(Debug, PartialEq)]
+enum DictionaryKey<'a> {
+    Constant(&'a Constant),
+    Variable(&'a str),
+}
+
+impl<'a> TryFrom<&'a Expr> for DictionaryKey<'a> {
+    type Error = ();
+
+    fn try_from(expr: &'a Expr) -> Result<Self, Self::Error> {
+        match &expr.node {
+            ExprKind::Constant { value, .. } => Ok(DictionaryKey::Constant(value)),
+            ExprKind::Name { id, .. } => Ok(DictionaryKey::Variable(id)),
+            _ => Err(()),
+        }
+    }
+}
+
+/// F601
+pub fn repeated_literal_keys(checker: &mut Checker, keys: &[Expr], values: &[Expr]) {
+    let num_keys = keys.len();
+    for i in 0..num_keys {
+        let k1: Option<DictionaryKey> = (&keys[i]).try_into().ok();
+        for j in i + 1..num_keys {
```

Comment:
This quadratic loop could be slow on a large dictionary. A `HashMap` would avoid that problem.

Reply:
Nice idea.

Reply:
Literally a leetcode problem and I failed it!

Reply:
@andersk - Do you think it's important to avoid removing duplicate keys with non-identical values?

Reply:
If we do, it's hard to avoid quadratic time in pathological cases because values aren't hashable. As an alternative, I could stringify them.

Reply:
We can make a wrapper type `struct ComparableExpr<'a>(&'a Expr)` that has custom `impl`s of `Eq` and `Hashable`.

Reply:
I did start on that, but `f64` itself isn't hashable.

Reply:
You can hash via [`f64::to_bits`](https://doc.rust-lang.org/std/primitive.f64.html#method.to_bits).

Reply:
Mmm ok, cool. I'll try that out. (Above, you're suggesting that `ComparableExpr` compares by ignoring the `Location` fields, right?)

Reply:
Right.

Reply:
This will come in a separate PR.